### PR TITLE
chore: update dependencies and use protobufjs

### PR DESCRIPTION
**Pull request by vasco-santos**
Updates dependencies based on the latest libp2p release. Most important, multiaddr and aegir were updated and we now use protobufjs instead of protons

Note that `src/proto.js` and `src/proto.d.ts` files were generated

BREAKING CHANGE: uses new multiaddr

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,8 +6,8 @@ stages:
   - cov
 
 node_js:
-  - '10'
-  - '12'
+  - '14'
+  - '16'
```
Comment (achingbrain) [inline .travis.yml:10]:
```suggestion
  - 'lts/*'
  - 'node'
```

File: package.json [hunk 2]
```diff
@@ -39,21 +41,23 @@
...
-    "debug": "^4.2.0",
+    "debug": "^4.3.1",
     "emittery": "^0.7.1",
-    "multiaddr": "^8.0.0",
-    "peer-id": "^0.14.2",
-    "protons": "^2.0.0"
+    "multiaddr": "^9.0.1",
+    "peer-id": "^0.14.8",
+    "protobufjs": "^6.11.2",
+    "protons": "^2.0.1"
```
Comment (achingbrain) [inline package.json:60]: Do we need `protobufjs` and `protons`?

**Review by achingbrain — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (vasco-santos) [inline package.json:60]:
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -39,21 +41,23 @@
...
   "dependencies": {
-    "debug": "^4.2.0",
+    "debug": "^4.3.1",
     "emittery": "^0.7.1",
-    "multiaddr": "^8.0.0",
-    "peer-id": "^0.14.2",
-    "protons": "^2.0.0"
+    "multiaddr": "^9.0.1",
+    "peer-id": "^0.14.8",
+    "protobufjs": "^6.11.2",
```
Comment (vasco-santos) [inline package.json:59]:
```suggestion
    "protobufjs": "^6.11.2"
```

File: package.json (hunk 2 above)
Comment (vasco-santos) [reply package.json:60]: no, I forgot to remove it 😓

**Review by achingbrain — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -24,12 +24,14 @@
...
   "engines": {
-    "node": ">=10.0.0",
+    "node": ">=14.0.0",
     "npm": ">=6.0.0"
   },
   "scripts": {
     "lint": "aegir lint",
-    "build": "aegir build",
+    "build": "npm run build:proto && npm run build:proto-types && aegir build --no-types",
+    "build:proto": "pbjs -t static-module -w commonjs -r libp2p-pubsub-discovery --force-number --no-verify --no-delimited --no-create --no-beautify --no-defaults --lint eslint-disable -o src/proto.js ./src/proto.proto",
```
Comment (achingbrain) [inline package.json:33]: Should this be `peer.proto` instead of `proto.proto`?

**Review by achingbrain — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (vasco-santos) [reply package.json:33]: sure, given that is a peer discovery piece, we should probably call it that way

**Pull request merged**